src/Time/Units.hs

```diff
@@ -136,6 +136,18 @@ instance KnownSymbol (ShowUnit unit) => Read (Time unit) where
             unless (timeUnitStr == symbolVal (Proxy @(ShowUnit unit))) pfail
             pure $ Time (n % m)
 
+instance Num (Time unit) where
```

Comment:
I think you can also add Haddock documentation to these manual instances. Like, say, that they behave just like derived ones, except `(*)` operator throws runtime error with `error`.